### ISSUE TITLE
Support workspace folders

### ISSUE THREAD

**Issue by mkaput** [starkware-libs/cairo#6127]
The project model should primarily be triggered by opening workspaces instead of opening files.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_workspaceFolders
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWorkspaceFolders

The ultimate goal would be that opening files won't be necessary in E2E tests.

Depends on #6124

**Comment by mkaput** [comment]
duplicate of #6229
